schema_based_on_diagram.sql

```diff
@@ -0,0 +1,74 @@
+CREATE TABLE
+    medical_histories (
+        id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+        admitted_at TIMESTAMP NOT NULL,
+        patient_id INT NOT NULL,
+        status VARCHAR(20) NOT NULL,
+        CONSTRAINT fk_patient_id FOREIGN KEY (patient_id) REFERENCES patients (id) ON DELETE CASCADE
+    );
+
+CREATE TABLE
+    patients (
+        id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+        name VARCHAR(50) NOT NULL,
+        date_of_birth DATE NOT NULL
+    );
+
+CREATE INDEX idx_medical_histories_patient_id ON medical_histories (patient_id);
+
+CREATE TABLE
+    invoices (
+        id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+        total_amount DECIMAL NOT NULL,
+        generated_at TIMESTAMP NOT NULL,
+        payed_at TIMESTAMP NOT NULL,
+        medical_history_id INT NOT NULL,
+        CONSTRAINT fk_medical_histories FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id)
+    );
+
+CREATE INDEX idx_invoices_medical_history_id ON invoices (medical_history_id);
+
+CREATE TABLE
+    invoice_items (
+        id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+        unit_price DECIMAL NOT NULL,
+        quantity INT NOT NULL,
+        total_price DECIMAL NOT NULL,
+        invoice_id INT NOT NULL,
+        treatment_id INT NOT NULL,
+        CONSTRAINT fk_invoices FOREIGN KEY (invoice_id) REFERENCES invoices (id),
+        CONSTRAINT fk_treatments FOREIGN KEY (treatment_id) REFERENCES treatments (id)
+    );
+
+CREATE TABLE
+    treatments (
+        id INT GENERATED ALWAYS AS IDENTITY,
+        type VARCHAR(20) NOT NULL,
+        name VARCHAR(50) NOT NULL,
+        PRIMARY KEY (id)
+    );
+
+CREATE INDEX idx_invoice_items_invoice_id ON invoice_items (invoice_id);
+
+CREATE INDEX idx_invoice_items_treatment_id ON invoice_items (treatment_id);
+
+ALTER TABLE treatments
+ADD COLUMN medical_history_id INTEGER;
+
+ALTER TABLE treatments ADD CONSTRAINT fk_medical_history FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id);
```

Comment:
👏 Great job on your progress so far! I appreciate your effort in experimenting with the addition of the `medical_history_id` in the `treatments` table and subsequently resolving the issue by creating a new table to follow the many-to-many relationship. However, here are a couple of suggestions:

- [ ] [Optional] To maintain clarity and avoid any confusion, it's recommended to remove any unnecessary code. Since you've removed the `medical_history_id` column in line number 69-70, please consider removing it from your SQL file as well to keep your codebase clean and more comprehensible.

- [ ] [Optioanl] Additionally, to enhance the usability of your SQL file, it would be beneficial to organize your code in the order you executed it. This will ensure that others can easily follow the logical progression of your SQL operations without encountering errors or confusion. By ordering your code effectively, you contribute to a more user-friendly and efficient codebase. Keep up the great work! 👍👨‍💻

Reply:
Thank you for your valuable feedback! I appreciate your attention to detail 🎯

Regarding the first suggestion, I understand the importance of keeping the codebase clean and comprehensible. However, in a previous code review, I was advised to leave queries that alter table structures in place for clarity. I agree that this specific case may seem a bit odd, as we declare `medical_history_id` and then drop it shortly after. I'll discuss this with the team to ensure we maintain consistency and clarity in our approach.

For the second point, if I understand correctly, you're suggesting that the index for `invoice_id` could have been created before working with the `treatments` table. I'll take a closer look at the order of operations and ensure that we follow a logical progression to enhance the SQL file's usability.

Thanks again for your insights, and I'm committed to improving our codebase for better collaboration and efficiency. 👍👨‍💻